Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -108,17 +119,21 @@ extension AlertStore {
 
     private func recordUpdateOfLatest(of identifier: Alert.Identifier,
```

Comment:
[nit] `recordUpdateOfLatest` is getting more complicated to handle the different use cases for `recordAcknowledgement` and `recordRetraction`. To me, it is getting harder to reason about the common function than if they were just implemented separately. Consider implementing separately.

Reply:
I'm sorry but I humbly disagree.  Separate functions would be largely duplicated code.  For example, it made it easier for you to add `purgeExpired`.  I'll try to make the code a little easier to read.

Reply:
Your call.